src/matlab2tikz.m

```diff
@@ -2930,15 +2939,13 @@ case size(cData,2) % specific y-data is given
 
             % Calculate the width of each bar and the center point shift as in
             % makebars.m
-            if isempty(m2t.barShifts)
-                % Get the shifts of the bar centers.
-                % In case of numBars==1, this returns 0,
-                % In case of numBars==2, this returns [-1/4, 1/4],
-                % In case of numBars==3, this returns [-1/3, 0, 1/3],
-                % and so forth.
-                barWidth = groupWidth/numBars; % assumption
-                m2t.barShifts = ((1:numBars) - 0.5) * barWidth - groupWidth/2;
-            end
+            % Get the shifts of the bar centers.
+            % In case of numBars==1, this returns 0,
+            % In case of numBars==2, this returns [-1/4, 1/4],
+            % In case of numBars==3, this returns [-1/3, 0, 1/3],
+            % and so forth.
+            barWidth = groupWidth/numBars; % assumption
+            barShift = (m2t.barplotId - 0.5) * barWidth - groupWidth/2;
```

Comment:
Yes, because `numBars` is actually the number of bar peers, i.e. how many bars per grouping. Previously, the shifts were stored globally and accessed later with `m2t.barShifts(m2t.barplotId)`. I just removed the shifts from the `m2t` object and calculate each individual shift locally.